apostle/mail.py

```diff
@@ -79,12 +83,18 @@ def to_recipient_dict(self):
 			'layout_id': self.layout_id,
 			'name': self.name,
 			'reply_to': self.reply_to,
-			'template_id': self.template_id
+			'template_id': self.template_id,
+            'attachments': self.attachments
 		}
 		for key in list(recipient_dict.keys()):
 			val = recipient_dict[key]
-			if val == None or val == {}:
+			if val == None or val == {} or val == []:
 				del recipient_dict[key]
 
 		return { self.email: recipient_dict }
 
+	def add_attachment(self, name, content):
+		self.attachments.append({
+			"name": name,
+			"data": base64.b64encode(content)
+		})
```

Comment:
Please ensure there's a trailing newline at the end of the files (for consistency).